alfresco-transform-aio/alfresco-transform-aio/src/main/java/org/alfresco/transformer/transformers/AllInOneTransformer.java

```diff
@@ -0,0 +1,135 @@
+ * (at your option) any later version.
+ * -
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ * -
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.transformer.transformers;
+
+import org.alfresco.transform.client.model.config.TransformConfig;
+import org.alfresco.transform.client.model.config.TransformOption;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Top level transformer managing multiple sub transformers.
+ *
+ * @author eknizat
+ */
+public class AllInOneTransformer implements Transformer
+{
+
+    private static final Logger logger = LoggerFactory.getLogger(AllInOneTransformer.class);
+
+    /**
+     * Represents the mapping between a transform and a transformer, multiple mappings can point to the same transformer.
+     */
+    private Map<String, Transformer> transformerTransformMapping = new HashMap();
+
+    public AllInOneTransformer()
+    {
+        // TODO - use observer style registration?
+        try
+        {
+            this.registerTransformer(new MiscAdapter());
+            this.registerTransformer(new TikaAdapter());
+        } catch (Exception e)
+        {
+            // rethrow as runtime exception, nothing else cane be done
+            throw new RuntimeException("Failed to register all transformers.", e);
+        }
+    }
+
+    /**
+     * The registration will go through all supported sub transformers and map them to the transformer implementation.
+     *
+     * @param transformer The transformer implementation,
+     *                    this could be a transformer managing multiple sub transformers.
+     * @throws Exception Exception is thrown if a mapping for a transformer name already exists.
+     */
+    public void registerTransformer(Transformer transformer) throws Exception
+    {
+        for (org.alfresco.transform.client.model.config.Transformer transformerConfig
+                : transformer.getTransformConfig().getTransformers())
+        {
+            String transformerName = transformerConfig.getTransformerName();
+            if (transformerTransformMapping.containsKey(transformerName))
+            {
+                throw new Exception("Transformer name " + transformerName + " is already registered.");
+            }
+
+            transformerTransformMapping.put(transformerName, transformer);
+            logger.debug("Registered transformer with name: '" + transformerName + "' ");
```

Comment:
Minor: surround with if debugEnabled (as per other parts of code) or else using logger with format args ?